vertx-web/src/main/java/io/vertx/ext/web/impl/RouteState.java

```diff
@@ -1068,16 +1068,18 @@ private boolean virtualHostMatches(HttpServerRequest request) {
     if (virtualHostPattern == null) {
       return true;
     }
+
     String host = request.host();
     if (host == null) {
       return false;
     }
-    for (String h : host.split(":")) {
-      if (virtualHostPattern.matcher(h).matches()) {
-        return true;
-      }
+
+    int portSeparatorIdx = host.lastIndexOf(':');
+    if (portSeparatorIdx > host.lastIndexOf(']')) {
```

Comment:
I think we should have some more explicit check here.

if `portSeparatorIdx` is -1 then `portSeparatorIdx` should be `host.length()`

IPv6 address is enclosed with `[ ... ]` so:

If the first char is `[` and the char before `portSeparatorIdx` is `]` then, the host string is the substring from `1` to `portSeparatorIdx - 1` otherwise is from `0` to `portSeparatorIdx`.

This way we can fight against attacks like:

```
Host: abc].com
```

Perhaps a safer bet would be to use the internal class `Origin` that already does most of this kind of validation.